prometheus-proc/src/Prometheus/Metric/Proc.hs

```diff
@@ -0,0 +1,175 @@
+{-# language ForeignFunctionInterface #-}
+{-# language OverloadedStrings #-}
+{-# language RecordWildCards #-}
+
+module Prometheus.Metric.Proc ( procMetrics ) where
+
+import Data.Char ( isSpace )
+import Data.Maybe ( catMaybes )
+import Data.String ( fromString )
+import Foreign.C
+import Prometheus
+import System.FilePath
+import System.IO.Unsafe
+import System.Posix.Memory ( sysconfPageSize )
+import System.Posix.Process ( getProcessID )
+import qualified Text.Regex.Applicative as RE
+import qualified Text.Regex.Applicative.Common as RE
+
+
+data ProcMetrics =
+  ProcMetrics
+
+
+procMetrics :: Prometheus.Metric ProcMetrics
+procMetrics =
+  Metric ( return ( ProcMetrics, collect ) )
+
+
+foreign import ccall unsafe
+  clk_tck :: CLong
+
+
+collect :: IO [ SampleGroup ]
+collect = do
+  pid <-
+    getProcessID
+
+  mprocStat <-
+      RE.match parseProcStat <$> readFile ( "/" </> "proc" </> show pid </> "stat" )
+
+  return
+    ( foldMap ( toMetrics ) mprocStat )
+
+
+toMetrics :: ProcStat -> [ SampleGroup ]
+toMetrics ProcStat{..} =
```

Comment:
I personally prefer `NamedFieldPuns` to `RecordWildcards`, as it helps me figure out where variables come from, and what exactly is in scope. YMMV though.

Reply:
Happy to go with whatever is preferred. I'm easy. I'll change over to that. Explicit is generally better than implicit.